x-pack/solutions/search/plugins/search_connectors/server/services/index.ts

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { PACKAGE_POLICY_SAVED_OBJECT_TYPE, PackagePolicy } from '@kbn/fleet-plugin/common';
+import { AgentPolicyServiceInterface, PackagePolicyClient } from '@kbn/fleet-plugin/server';
+import type { Logger, SavedObjectsClientContract } from '@kbn/core/server';
+import { NATIVE_CONNECTOR_DEFINITIONS, fetchConnectors } from '@kbn/search-connectors';
+
+export interface ConnectorMetadata {
+  id: string;
+  name: string;
+  service_type: string;
+}
+
+export interface PackagePolicyMetadata {
+  package_policy_id: string;
+  agent_policy_ids: string[];
+  connector_metadata: ConnectorMetadata;
+}
+
+const connectorsInputName = 'connectors-py';
+const pkgName = 'elastic_connectors';
+const pkgVersion = '0.0.4';
```

Comment:
do we need this version hardcoded here? The current (latest) version in the integration registry should be def tracked somewhere by fleet, can we look it up in the package registry dynamically?

Context, `0.0.4` is already [outdated](https://github.com/elastic/integrations/blob/main/packages/elastic_connectors/changelog.yml)

Reply:
Maybe code edited in this PR will help? https://github.com/elastic/kibana/pull/192081/files here I was able to access package info and adjust permissions dynamically